test/files/run/t11736.scala

```diff
@@ -0,0 +1,19 @@
+// scalac: -Xelide-below ASSERTION
+
+import scala.tools.partest.BadRunTest
+
+trait Foo {
+  trait Bar {
+    assert(Foo.this ne null)        // fails non-trivially
+  }
+}
+
+abstract class Foo1 extends Foo
+
+object Foo2 extends Foo1 {
+  new Bar {}
+}
+
+object Test extends BadRunTest {
+  def run() = run(assertsOn)(Foo2: Unit)
```

Comment:
I find this very convoluted, and/or I don't really understand the abstraction here... Could we write the test wihout the indirections and special handling of assertions?

Reply:
I just want to ensure that my vanilla `assert` is not elided. But for `BadRun`, I considered that `assertFails` or `assertThrows` suffices. Because of java assert, I am less prepared to believe the condition will be asserted. As opposed to unit test assertions. But given your impression, I'll just drop it.